### PR TITLE
GNMI Performance test framework

### PR DESCRIPTION
**Pull request by ganglyu**
#### Why I did it
Generate test data to verify GNMI performance.
Create test framework to verify GNMI performance.

#### How I did it
Add script to generate DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE.
Add unit test to get the time for parsing json, yang validation and swsscommon.

#### How to verify it
Run unit for sonic-gnmi.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by prsunny** [comment]
@ganglyu , could you provide some numbers on performance with this test as reference?

**Comment by ganglyu** [comment]
@prsunny 
Both DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE have 1000 entries. Parse input request takes 3 ms, yang validation takes about 250 ms, swsscommon takes about 250 ms. And we don't have DASH yang models for now.
    client_test.go:375: Parse request: Route table takes time: 2.994721ms
    client_test.go:375: Parse request, yang validation: Route table takes time: 260.543897ms
    client_test.go:375: Parse request, yang validation and swsscommon: Route table takes time: 510.016917ms
    client_test.go:375: Parse request: Mapping table takes time: 3.740626ms
    client_test.go:375: Parse request, yang validation: Mapping table takes time: 248.198111ms
    client_test.go:375: Parse request, yang validation and swsscommon: Mapping table takes time: 482.138824ms

**Comment by prsunny** [comment]
> @prsunny Both DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE have 1000 entries. Parse input request takes 3 ms, yang validation takes about 250 ms, swsscommon takes about 250 ms. And we don't have DASH yang models for now. client_test.go:375: Parse request: Route table takes time: 2.994721ms client_test.go:375: Parse request, yang validation: Route table takes time: 260.543897ms client_test.go:375: Parse request, yang validation and swsscommon: Route table takes time: 510.016917ms client_test.go:375: Parse request: Mapping table takes time: 3.740626ms client_test.go:375: Parse request, yang validation: Mapping table takes time: 248.198111ms client_test.go:375: Parse request, yang validation and swsscommon: Mapping table takes time: 482.138824ms

@ganglyu , what is swsscommon here? Is it writing to APP_DB?

**Comment by qiluo-msft** [comment]
Change file extension to ".json" ?

---
Refers to: testdata/dash_mapping_batch.txt:1 in e62f9d7. [](commit_id = e62f9d763c1d87cf0c7848122379795fcc1f8a60, deletion_comment = False)

---
In reply to: [1374268546](https://github.com/nvidia-sonic/sonic-gnmi/pull/10#issuecomment-1374268546) [](http://example.com/codeflow?ancestors=1374268546)

**Comment by qiluo-msft** [comment]
Possible to prettier this json content?

---
Refers to: testdata/dash_mapping_batch.txt:1 in e62f9d7. [](commit_id = e62f9d763c1d87cf0c7848122379795fcc1f8a60, deletion_comment = False)

---
In reply to: [1374268766](https://github.com/nvidia-sonic/sonic-gnmi/pull/10#issuecomment-1374268766) [](http://example.com/codeflow?ancestors=1374268766)

File: gnmi_server/server_test.go [hunk 1]
```diff
@@ -2738,78 +2738,6 @@ func TestClient(t *testing.T) {
     s.s.Stop()
 }
 
-func TestGnmiSetBatch(t *testing.T) {
```
Comment (qiluo-msft) [inline gnmi_server/server_test.go]: Why delete testcase?

Comment (ganglyu) [reply gnmi_server/server_test.go]: We need to verify performance for DASH_ROUTE_TABLE and DASH_VNET_MAPPING_TABLE.
TestGnmiSetBatch used DASH_QOS for batch test, it does not meet our requirement.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**